framework/db/Command.php

```diff
@@ -736,7 +736,110 @@ public function dropIndex($name, $table)
     {
         $sql = $this->db->getQueryBuilder()->dropIndex($name, $table);
 
-        return $this->setSql($sql);
+        return $this->setSql($sql)->requireTableSchemaRefresh($table);
+    }
+
+    /**
+     * Creates a SQL command for adding an unique constraint to an existing table.
+     * @param string $name the name of the unique constraint.
+     * The name will be properly quoted by the method.
+     * @param string $table the table that the unique constraint will be added to.
+     * The name will be properly quoted by the method.
+     * @param string|array $columns the name of the column to that the constraint will be added on.
+     * If there are multiple columns, separate them with commas.
+     * The name will be properly quoted by the method.
+     * @return $this the command object itself.
+     * @since 2.0.13
+     */
+    public function addUnique($name, $table, $columns)
```

Comment:
It sounds reasonable, but here are thoughts behind `addUnique`:
1. Almost all `Command` / `QueryBuilder` names follow SQL commands.
2. We `CREATE INDEX`es but we `ADD CONSTRAINT`s when we `ALTER TABLE`.
3. `Unique` isn't ideal, but why don't we use `Constraint` suffix elsewhere then (hint: BC)? I prefer to stick with consistency - like @dynasource said.

But I'm not principled in this case.